capsule/src/main/java/Capsule.java

```diff
@@ -3393,7 +3398,8 @@ private String processOutgoingPath(Object p) {
     /////////// JAR Extraction ///////////////////////////////////
     private static void extractJar(JarInputStream jar, Path targetDir) throws IOException {
         for (JarEntry entry; (entry = jar.getNextJarEntry()) != null;) {
-            if (entry.isDirectory() || !shouldExtractFile(entry.getName()))
+            if (entry.isDirectory() || !shouldExtractFile(entry.getName())
+                    || targetDir.resolve(entry.getName()).toFile().exists())
```

Comment:
Please use the new file API (`Files.exists`). The old file API is incompatible with new libraries (like Google's jimfs, which is used in the tests and should be avoided in general...